lnwallet/channel.go

```diff
@@ -240,8 +245,15 @@ func (c *commitment) toChannelDelta() (*channeldb.ChannelDelta, error) {
 	// transaction.
 	locateOutputIndex := func(p *PaymentDescriptor) uint16 {
 		var idx uint16
+
+		var pkScript []byte
+		if ourCommit {
+			pkScript = p.ourPkScript
+		} else {
+			pkScript = p.theirPrevPkScript
```

Comment:
The reason for using `theirPrevPkScript` is that we will overwrite `theirPkScript` with a new value before we have a chance to call `toChannelDelta()` for the remotes revocation log. When we call `SignNextCommitment()` we will overwrite `theirPkScript` in `addHTLC()`, which is why we save the previous pkScript as `theirPrevPkScript` so that we have it available for when `toChannelDelta()` is to called when we receive a revocation from the remote peer later.

When setting `[our|their]pkScripts` in `addHTLC()` we are accounting for whether it is an incoming or outgoing htlc when we generate the scripts. I don't see how it is necessary to check for this in `locateOutputIndex()`. We should only have to check if we are operating on the local or remote commitmentchain, which is the same throughout the call to `toChannelDelta()`.